index.js

```diff
@@ -17,7 +17,8 @@ http.request = function (params, cb) {
     }
     if (!params) params = {}
     if (!params.host && !params.port) {
-        params.port = defaults.port
+        if (params.protocol === 'https:') params.port = 443
+        else params.port = defaults.port
```

Comment:
can u update `defaults` to be a lookup by `protocol` instead? otherwise we have two places with magic values